### ISSUE TITLE
[Feature]: Support IAM access for S3 storage

### ISSUE THREAD

**Issue by lilida** [milvus-io/milvus#16898]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

Right now, S3 access must be done by setting access id and secret key using milvus-operator. These sensitive information are passed as environment var and if not set, the milvus-operator won't be able to create the deployment.

### Describe the solution you'd like.

Support S3 access via IAM. In MilvuCluster spec, it should allow user don't set and SecretRef and allow configure service account for the pod.

**Comment by xiaofan-luan** [comment]
any ideas?

**Comment by haorenfsa** [comment]
@lilida  Thx for the feadback.

> `These sensitive information are passed as environment var and if not set`

You can create a `Secret` of the AK & SK credentials. And use `secretRef` in the `MilvusCluster` spec.


> `Support S3 access via IAM. In MilvuCluster spec, it should allow user don't set and SecretRef and allow configure service account for the pod.`

I don't quite get it. There seems to be some typos.  Could you describe it more specifically? 

If I get it right, you want the `milvus-operator` to do the secret configuration instead of user. Then the `milvus-operator` has to hold some admin account of aws in order to acquire or assign AccessKeys, which seems to be out of the scope of what an operator should do.

**Comment by haorenfsa** [comment]
> https://app.slack.com/client/TM98LD8DQ/CMNHUC371/thread/CMNHUC371-1652138542.451079

Oh I got the context above

**Comment by haorenfsa** [comment]
I'll check if minio sdk supports `IAM role via an OIDC web identity token file` first.

**Comment by haorenfsa** [comment]
`minio-go` already supported:  https://github.com/minio/minio-go/issues/1156

**Comment by lilida** [comment]
@haorenfsa Thanks for looking into this promptly. Does this mean milvus it self actually support this but what it is lack is the way to configure this in milvus-operator today?

**Comment by haorenfsa** [comment]
> @haorenfsa Thanks for looking into this promptly. Does this mean milvus it self actually support this but what it is lack is the way to configure this in milvus-operator today?

Nope, it still needs to support config s3 client to use [credential with IAM](https://pkg.go.dev/github.com/minio/minio-go/pkg/credentials#Credentials)

**Comment by haorenfsa** [comment]
can be closed
